qpid/cpp/src/qpid/broker/amqp/Outgoing.cpp

```diff
@@ -79,6 +79,9 @@ OutgoingFromQueue::OutgoingFromQueue(Broker& broker, const std::string& source,
 
 void OutgoingFromQueue::init()
 {
+    // observe queue for changes to track undeliverable messages
+    queue->getObservers().add(boost::shared_ptr<QueueObserver>(shared_from_this()));
+
```

Comment:
sure that sounds reasonable. you think its enough to just register it during `handle` below, with a boolean flag on the class?